src/implementation.js

```diff
@@ -1,18 +1,86 @@
 'use strict';
 
-/* Example singleton with es6
- * let myInstance = null;
- * const SomeClass = require('SomeClass');
- *
- * module.exports = function() {
- *     if (!myInstance) {
- *         myInstance = new SomeClass();
- *     }
- *
- *     return myInstance;
- * }
- *
-**/
-
-//export result here
-module.exports = {};
+const Templater = require('toki-templater');
+const Request = require('superagent');
+const Joi = require('joi');
+const Promise = require('bluebird');
+
+const configSchema = Joi.object().keys({
+    name: Joi.string(),
+    url: Joi.string().uri().required(),
+    passThroughHeaders: Joi.alternatives().try(Joi.boolean(), Joi.array()),
+    headers: Joi.object(),
+    payload: Joi.any(),
+    method: Joi.string().allow(['get', 'post', 'put', 'del', 'patch']).default('get'),
+    type: Joi.string().allow(['json', 'application/json']),
+    clientResponse: Joi.alternatives().try(Joi.boolean(), Joi.object()),
+    output: Joi.any()
+});
+
+module.exports = function () {
+
+    return Promise
+    .resolve()
+    .bind(this)
+    .then( () => Templater(this.action, configSchema, { allowUnknown: true } ) )
+    .then((config) => {
+
+        this.config = config;
+    })
+    .then(() => {
+        //url
+        const req = Request(this.config.method, this.config.url);
+
+        //content type
+        if (this.config.type) {
+            req.type(this.config.type);
+        }
+
+        //body
+        if (this.config.payload) {
+            req.send(this.config.payload);
```

Comment:
Not with the way the bridge is written because we handle the request lifetime more flexibly than hapi does.